integration/build/build_test.go

```diff
@@ -136,6 +137,56 @@ func buildContainerIdsFilter(buildOutput io.Reader) (filters.Args, error) {
 	}
 }
 
+// TestBuildMultiStageCopy verifies that copying between stages works correctly
+// regression test for docker/for-win#4349, ENGCORE-935, where creating the target
+// directory failed on Windows, because `os.MkdirAll()` was called with a volume
+// GUID path (\\?\Volume{dae8d3ac-b9a1-11e9-88eb-e8554b2ba1db}\newdir\hello}),
+// which currently isn't supported by Golang.
+func TestBuildMultiStageCopy(t *testing.T) {
+	ctx := context.Background()
+
+	f, err := filepath.Abs("testdata/Dockerfile." + t.Name())
```

Comment:
Maybe inline it as in other tests?

Reply:
That didn't work; looks like on Windows the CWD is different, and it sees `C:\gopath\src\github.com\docker\docker\` as current directory, not `C:\gopath\src\github.com\docker\docker\integration\build\`.

I can work around that, but looking _why_

CI is running the integration tests from here;
https://github.com/moby/moby/blob/4fb5e9e696fb50d65966a4b6b3c7906046d7c6a4/hack/make.ps1#L338-L354

Which looks quite similar to the Linux equivalent;

https://github.com/moby/moby/blob/4fb5e9e696fb50d65966a4b6b3c7906046d7c6a4/hack/make/.integration-test-helpers#L44-L48 


I wonder if `System.Diagnostics.ProcessStartInfo` inherits the working directory, or if it needs to be set explicitly 🤔

Reply:
Looks like that may be it: it has a `ProcessStartInfo.WorkingDirectory` property

From the documentation; https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.workingdirectory?view=netframework-4.8

> When `UseShellExecute` is `true`, the fully qualified name of the directory that contains
> the process to be started. When the `UseShellExecute` property is `false`, the working
> directory for the process to be started. The default is an empty string (`""`).

Reply:
Opened https://github.com/moby/moby/pull/39698 to fix that

I'll rebase this PR on top of that one to get the fix in this branch